deploy/common/monitoring.sh

```diff
@@ -2,6 +2,8 @@
 
 kubectl -n stackrox patch svc/sensor -p '{"spec":{"ports":[{"name":"monitoring","port":9090,"protocol":"TCP","targetPort":9090}]}}'
 kubectl -n stackrox patch svc/central -p '{"spec":{"ports":[{"name":"monitoring","port":9090,"protocol":"TCP","targetPort":9090}]}}'
+kubectl -n stackrox patch daemonset/collector --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/1/ports", "value":[{"containerPort":9091,"name":"cmonitor","protocol":"TCP"}]}]'
+kubectl -n stackrox set env daemonsets/collector --containers="compliance" ROX_METRICS_PORT=":9091"
```

Comment:
I think this line is note executed in CI and the `gke-nongroovy-e2e-tests` try to pull up the compliance with monitoring on port 9090 which results in a panic:

```
No certificates found in /usr/local/share/ca-certificates
No certificates found in /etc/pki/injected-ca-trust
main: 2023/01/30 13:22:25.896767 main.go:253: Info: Running StackRox Version: 3.73.x-588-g8bbc45ad6a
main: 2023/01/30 13:22:25.897803 main.go:265: Info: Initialized Sensor gRPC stream connection
pkg/metrics: 2023/01/30 13:22:25.900635 http_server.go:64: Panic: Unexpected termination of metrics HTTP server: listen tcp :9090: bind: address already in use
panic: Unexpected termination of metrics HTTP server: listen tcp :9090: bind: address already in use

goroutine 13 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0008bc2c0, {0x0, 0x0, 0x0})
	go.uber.org/zap@v1.23.0/zapcore/entry.go:230 +0x486
go.uber.org/zap.(*SugaredLogger).log(0xc000850008, 0x4, {0x2169e65?, 0x0?}, {0xc00008bfa8?, 0x0?, 0x0?}, {0x0, 0x0, 0x0})
	go.uber.org/zap@v1.23.0/sugar.go:234 +0x13b
go.uber.org/zap.(*SugaredLogger).Panicf(...)
	go.uber.org/zap@v1.23.0/sugar.go:159
github.com/stackrox/rox/pkg/metrics.runForever(0x0?)
	github.com/stackrox/rox/pkg/metrics/http_server.go:64 +0x71
created by github.com/stackrox/rox/pkg/metrics.(*HTTPServer).RunForever
	github.com/stackrox/rox/pkg/metrics/http_server.go:58 +0x1e5
```

Reply:
Ah, thank you for the investigation!
I have moved the port definition into the DaemonSet itself by default, as we always want to have a different port for the compliance metrics.